pipelines/foreman_split_puppet.yml

```diff
@@ -0,0 +1,109 @@
+
+- hosts:
+  - "{{ forklift_server_name }}"
+  - "{{ forklift_proxy_name }}"
+  - "{{ forklift_puppet_name }}"
+  become: yes
+  vars:
+    foreman_installer_skip_installer: true
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: foreman_server_repositories
+    - role: update_os_packages
+    - role: foreman_installer
+
+- hosts: "{{ forklift_puppet_name }}"
+  become: yes
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: foreman_installer
+      foreman_installer_options_internal_use_only:
+        - "--no-enable-foreman"
+        - "--no-enable-foreman-cli"
+        - "--no-enable-foreman-proxy"
+        - "--puppet-server true"
+        - "--puppet-server-foreman-url https://{{ forklift_server_name }}.{{ ansible_domain }}"
+        - "--puppet-server-admin-api-whitelist localhost"
+        - "--puppet-server-admin-api-whitelist {{ forklift_puppet_name }}.{{ ansible_domain }}"
+        - "--puppet-server-admin-api-whitelist {{ forklift_proxy_name }}.{{ ansible_domain }}"
+        - "--puppet-server-ca-client-whitelist localhost"
+        - "--puppet-server-ca-client-whitelist {{ forklift_puppet_name }}.{{ ansible_domain }}"
+        - "--puppet-server-ca-client-whitelist {{ forklift_proxy_name }}.{{ ansible_domain }}"
+        - "--puppet-autosign-entries {{ forklift_server_name }}.{{ ansible_domain }}"
+        - "--puppet-autosign-entries {{ forklift_proxy_name }}.{{ ansible_domain }}"
+
+# TODO: set foreman's trusted hosts to [{{ forklift_puppet_name }}.{{ ansible_domain }}]
+- hosts: "{{ forklift_server_name }}"
+  become: yes
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: puppet_agent
+      puppet_agent_bootstrap_server: "{{ forklift_puppet_name }}.{{ ansible_domain }}"
+    - role: foreman_installer
+      foreman_installer_options_internal_use_only:
+        - "--enable-foreman"
+        - "--foreman-initial-admin-password changeme"
+        - "--foreman-oauth-consumer-key oauth-key"
+        - "--foreman-oauth-consumer-secret oauth-secret"
+        - "--enable-foreman-cli"
+        - "--no-enable-foreman-proxy"
+        - "--no-enable-puppet"
+
+- hosts: "{{ forklift_proxy_name }}"
+  become: yes
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: puppet_agent
+      puppet_agent_bootstrap_server: "{{ forklift_puppet_name }}.{{ ansible_domain }}"
+    - role: foreman_installer
+      foreman_installer_options_internal_use_only:
+        - "--no-enable-foreman"
+        - "--no-enable-foreman-cli"
```

Comment:
Great use case for getting https://github.com/theforeman/foreman-installer/pull/262 into the repository

Reply:
Agreed, but that requires more changes. Currently no foreman documentation mentions the `--scenario` parameter so we need to go over it all if we start to ship it in `foreman-installer`. However, as we add more of these it's starting to make more sense. https://theforeman.org/manuals/1.23/index.html#3.2.3InstallationScenarios basically repeat a lot of these options.